File: scripts/smoke-dashboards.mjs
// Smoke test: load new role dashboards, assert no console/page errors.
// Usage: node scripts/smoke-dashboards.mjs
import { chromium } from "playwright";

const BASE = process.env.SMOKE_BASE_URL ?? "http://localhost:8080";
const ROUTES = ["/dashboard/developer", "/dashboard/dev-manager", "/dashboard/promise-tracker"];

const results = [];
const browser = await chromium.launch({ headless: true });
try {
  for (const path of ROUTES) {
    const ctx = await browser.newContext({ viewport: { width: 1280, height: 900 } });
    const page = await ctx.newPage();
    const errors = [];
    page.on("console", (m) => { if (m.type() === "error") errors.push(m.text()); });
    page.on("pageerror", (e) => errors.push(String(e.message ?? e)));
    const resp = await page.goto(BASE + path, { waitUntil: "domcontentloaded", timeout: 30000 });
    await page.waitForLoadState("networkidle", { timeout: 15000 }).catch(() => {});
    const status = resp?.status() ?? 0;
    const hasSidebar = await page.locator('aside').first().isVisible().catch(() => false);
    results.push({ path, status, errors, hasSidebar });
    await ctx.close();
  }
} finally {
  await browser.close();
}

let failed = 0;
for (const r of results) {
  const ok = r.status === 200 && r.errors.length === 0 && r.hasSidebar;
  if (!ok) failed++;
  console.log(`${ok ? "PASS" : "FAIL"}  ${r.path}  status=${r.status}  errors=${r.errors.length}  sidebar=${r.hasSidebar}`);
  for (const e of r.errors) console.log("   •", e);
}
process.exit(failed ? 1 : 0);

File: scripts/smoke-dashboards.py
"""Smoke test: load new role dashboards and assert no console/page errors.

Usage:  python3 scripts/smoke-dashboards.py
Env:    SMOKE_BASE_URL (default http://localhost:8080)
"""
import asyncio, os, sys
from playwright.async_api import async_playwright

BASE = os.environ.get("SMOKE_BASE_URL", "http://localhost:8080")
ROUTES = ["/dashboard/developer", "/dashboard/dev-manager", "/dashboard/promise-tracker"]

async def check(browser, path):
    ctx = await browser.new_context(viewport={"width": 1280, "height": 900})
    page = await ctx.new_page()
    errors = []
    page.on("console", lambda m: errors.append(m.text) if m.type == "error" else None)
    page.on("pageerror", lambda e: errors.append(str(e)))
    resp = await page.goto(BASE + path, wait_until="domcontentloaded", timeout=30000)
    try:
        await page.wait_for_load_state("networkidle", timeout=15000)
    except Exception:
        pass
    status = resp.status if resp else 0
    try:
        sidebar = await page.locator("aside").first.is_visible()
    except Exception:
        sidebar = False
    await ctx.close()
    return {"path": path, "status": status, "errors": errors, "sidebar": sidebar}

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        try:
            results = [await check(browser, r) for r in ROUTES]
        finally:
            await browser.close()
    failed = 0
    for r in results:
        ok = r["status"] == 200 and not r["errors"] and r["sidebar"]
        failed += 0 if ok else 1
        print(f"{'PASS' if ok else 'FAIL'}  {r['path']}  status={r['status']}  errors={len(r['errors'])}  sidebar={r['sidebar']}")
        for e in r["errors"]:
            print("   \u2022", e)
    sys.exit(1 if failed else 0)

asyncio.run(main())